folly/Unicode.cpp

```diff
@@ -94,7 +94,7 @@ char32_t utf8ToCodePoint(
 
   fst <<= 1;
 
-  for (unsigned int i = 1; i != 3 && p + i < e; ++i) {
+  for (unsigned int i = 1; i != 4 && p + i < e; ++i) {
     unsigned char tmp = p[i];
 
     if ((tmp & 0xC0) != 0x80) {
```

Comment:
Github won't let me add the comment there, but does this also need to check for surrogates (see line 120)